DQM/SiTrackerPhase2/plugins/Phase2ITMonitorCluster.cc

```diff
@@ -0,0 +1,416 @@
+  std::string top_folder = config_.getParameter<std::string>("TopFolderName");
+  std::stringstream folder_name;
+
+  ibooker.cd();
+  folder_name << top_folder << "/";
+  ;
+  ibooker.setCurrentFolder(folder_name.str());
+
+  edm::LogInfo("Phase2ITMonitorCluster") << " Booking Histograms in: " << folder_name.str();
+  std::stringstream HistoName;
+
+  HistoName << "NumberClusters";
+  numberClusters_ = phase2tkutil::book1DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalNClusters"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_ClusterPosition_XY_IT_barrel";
+  globalXY_barrel_ = phase2tkutil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionXY_PXB"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_ClusterPosition_RZ_IT_barrel";
+  globalRZ_barrel_ = phase2tkutil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionRZ_PXB"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_ClusterPosition_XY_IT_endcap";
+  globalXY_endcap_ = phase2tkutil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionXY_PXEC"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_ClusterPosition_RZ_IT_endcap";
+  globalRZ_endcap_ = phase2tkutil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionRZ_PXEC"), HistoName.str(), ibooker);
+
+  //Now book layer wise histos
+  edm::ESWatcher<TrackerDigiGeometryRecord> theTkDigiGeomWatcher;
+  if (theTkDigiGeomWatcher.check(iSetup)) {
+    edm::ESHandle<TrackerGeometry> geomHandle = iSetup.getHandle(geomToken_);
+    for (auto const& det_u : tkGeom_->detUnits()) {
+      //Always check TrackerNumberingBuilder before changing this part
+      if (!(det_u->subDetector() == GeomDetEnumerators::SubDetector::P2PXB ||
+            det_u->subDetector() == GeomDetEnumerators::SubDetector::P2PXEC))
+        continue;
+      unsigned int detId_raw = det_u->geographicalId().rawId();
+      edm::LogInfo("Phase2ITMonitorRecHit") << "Detid:" << detId_raw << "\tsubdet=" << det_u->subDetector()
+                                            << "\t key=" << phase2tkutil::getITHistoId(detId_raw, tTopo_) << std::endl;
+      bookLayerHistos(ibooker, detId_raw, top_folder);
+    }
+  }
+}
+
+//////////////////Layer Histo/////////////////////////////////
+void Phase2ITMonitorCluster::bookLayerHistos(DQMStore::IBooker& ibooker, uint32_t det_id, std::string& subdir) {
+  std::string folderName = phase2tkutil::getITHistoId(det_id, tTopo_);
+  if (folderName.empty())
+    return;
+
+  std::map<std::string, ClusterMEs>::iterator pos = layerMEs_.find(folderName);
+
+  if (pos == layerMEs_.end()) {
+    ibooker.cd();
+    ibooker.setCurrentFolder(subdir + "/" + folderName);
+
+    edm::LogInfo("Phase2ITMonitorCluster") << " Booking Histograms in: " << subdir + "/" + folderName;
+
+    std::ostringstream histoName;
```

Comment:
why not a std::string?

Reply:
This is changed now. will push the updated code after test.